File: octopoints_flutter/lib/service/impl/match_service_impl.dart
import 'package:octopoints_flutter/db/database.dart';
import 'package:octopoints_flutter/service/service.dart';

class MatchServiceImpl extends OctopointsService implements MatchService {
  @override
  Future<Match> createMatch(Match match) async {
    int id = await db.matchDao.create(match.toMatchEntity());
    return match.setId(id);
  }

  @override
  Future<int> deleteMatch(Match match) {
    return db.matchDao.remove(match.toMatchEntity());
  }

  @override
  Future<List<Match>> getMatches() async {
    List<Match> matches = [];

    for (MatchEntity entity in await db.matchDao.getMatches()) {
      Match match = entity.toMatchModel();
      match.rule =
          await OctopointsService.ruleService.getRuleByMatchId(match.id);
      matches.add(match);
    }

    return matches;
  }
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/impl/MatchServiceImpl.kt
package dev.tpavani.octopoints.service.impl

import dev.tpavani.octopoints.db.DbSingleton
import dev.tpavani.octopoints.db.entity.MatchEntity
import dev.tpavani.octopoints.service.MatchService
import dev.tpavani.octopoints.service.model.Match
import dev.tpavani.octopoints.service.utils.ModelMapper

class MatchServiceImpl: MatchService {
    override suspend fun createMatch(match: Match): Match {
        var id: Int = DbSingleton.getDb().matchDao().create(ModelMapper.toMatchEntity(match))
        return match.setId(id)
    }

    override suspend fun deleteMatch(match: Match) {
        DbSingleton.getDb().matchDao().remove(ModelMapper.toMatchEntity(match))
    }

    override suspend fun getMatcher(): List<Match> {
        var list: MutableList<Match> = mutableListOf()
        for (matchEntity in DbSingleton.getDb().matchDao().getMatches()){
            var match = ModelMapper.toMatchModel(matchEntity)
            match.rule = ModelMapper.toRuleModel(DbSingleton.getDb().ruleDao().getRule(match.getId()))
            list.add(match)
        }
        return list
    }
}